core/src/main/java/org/apache/carbondata/core/datamap/dev/DataMap.java

```diff
@@ -26,13 +28,6 @@
 public interface DataMap {
 
   /**
```

Comment:
I think DataMapWriter should belong to DataMap, user should get writer from here. No need of seperate register for DataMapWriter as already DataMapStoreManager manages it